hls-plugin-api/src/Ide/Types.hs

```diff
@@ -76,9 +109,15 @@ import           Text.Regex.TDFA.Text            ()
 
 -- ---------------------------------------------------------------------
 
-newtype IdePlugins ideState = IdePlugins
-  { ipMap :: [(PluginId, PluginDescriptor ideState)]}
-  deriving newtype (Monoid, Semigroup)
+newtype IdePlugins ideState = IdePlugins_ { ipMap_ :: HashMap PluginId (PluginDescriptor ideState)}
+  deriving newtype (Semigroup, Monoid)
+
+-- | Smart constructor that deduplicates plugins
```

Comment:
Is this comment accurate now? Or does it now give them in priority order?

Reply:
It still deduplicates. The comment is correct but incomplete